src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -18,42 +18,43 @@ import {
 import { ExpressionTransformFunction, transformComputedFieldsQuery } from 'core/query-builder.utils';
 import { AssetCalibrationFieldNames } from './constants';
 import { AssetModel, AssetsResponse } from 'datasources/asset-common/types';
-import TTLCache from '@isaacs/ttlcache';
-import { metadataCacheTTL } from 'datasources/data-frame/constants';
 import { SystemMetadata } from 'datasources/system/types';
 import { defaultOrderBy, defaultProjection } from 'datasources/system/constants';
 import { QueryBuilderOperations } from 'core/query-builder.constants';
 import { Workspace } from 'core/types';
 import { parseErrorMessage } from 'core/errors';
+import { Subject } from 'rxjs';
```

Comment:
@saradei-ni I don't think we should use rxjs with React. 

RxJS is not an idiomatic way to manage ui state changes within React

Reply:
Please use hooks instead. If needed, you can also implement a custom hook and trigger rendering when the map is updated

Reply:
u can also add a useState hook and useRef hooks to your maps to avoid rerendering on change of the map and trigger only when the entire dependency set is loaded

Reply:
I have updated the approach to use promises instead of rxjs. The resolve will be triggered after loading is complete. The editor can set the state afterwards knowing the loading has been done, it waits for both promises to resolve.

This reactive approach ensures the editor is notified when the loading is complete. The previous approach wasn't consistent as the datasource object is only loaded once, and the async changes wouldn't properly reflect in editor when the flags were changed. Also, this covers the case when a query is not made at init, and the dependencies wouldn't load.

Let me know if we need any additional changes.